components/places/src/storage/bookmarks.rs

```diff
@@ -425,8 +380,14 @@ fn delete_bookmark_in_tx(db: &PlacesDb, guid: &SyncGuid) -> Result<bool> {
             return Ok(false);
         }
     };
+    // There's an argument to be made here that we should still honor the
+    // deletion in the case of this corruption, since it would be fixed by
+    // performing the deletion, and the user wants it gone...
```

Comment:
Sync should handle this correctly during merging, so this (writing a local tombstone without bumping the parent's change counter, since we don't have a parent) is most likely safe to do for synced and non-synced bookmarks, but let's double-check.